geth/jail/internal/loop/loop.go

```diff
@@ -40,6 +42,27 @@ type Loop struct {
 	lock  sync.RWMutex
 	tasks map[int64]Task
 	ready chan Task
+	state
+}
+
+// state encapsulates the loop's open and closedness. This is a boolean variable
```

Comment:
Not a big one, but could you move the definition of `state` in front of its usage in `Loop`? Maybe I'm too much grown up with Pascal teaching this order. 😁